samcli/lib/package/artifact_exporter.py

```diff
@@ -616,11 +614,17 @@ def export(self):
             return self.template_dict
 
         self.template_dict = self.export_global_artifacts(self.template_dict)
+        global_function_properties = self.template_dict.get("Globals", {}).get("Function", {})
 
         for resource_id, resource in self.template_dict["Resources"].items():
 
             resource_type = resource.get("Type", None)
-            resource_dict = resource.get("Properties", None)
+            if global_function_properties:
+                # Merging global and local Function properties. Local will override global.
+                resource_dict = {**global_function_properties, **resource.get("Properties", {})}
```

Comment:
Hang on. You don't need to do this here. Just run SAM Translator to merge Globals for you before package command starts. This is a more scalable approach because there are other parts of SAM template that you might need to "merge" before packaging.

This functionality should be accessible through SamFunctionProvider. Look at the InvokeContext code paths to get an idea.

Reply:
We might need to think deeper here. While I agree what @sanathkr is suggesting Is the right thing to do, SAM Translator requires additional permissions, which some depend on the resources in the template (SAR). Ideally, we should be able to do a dry run of the transform without needing permissions but that is way more work. So we should make sure adding additional permissions on package is something we want to support long term.

Reply:
But how do we run local invoke without these permissions?

Reply:
Fair. Validate has those requirements. Ignore me and my memory :)

Reply:
The question to ask is, did package have those permissions before? If we are adding it now, does it mean the permissions required to package change, and hence breaking tighlty scoped roles used in CI/CD for sam package, sam deploy?

Reply:
> Hang on. You don't need to do this here. Just run SAM Translator to merge Globals for you before package command starts. 

Updated code to use SamFunctionProvider before exporting Resources.